core/arch/arm/kernel/user_ta.c

```diff
@@ -437,6 +438,68 @@ static void user_ta_dump_state(struct tee_ta_ctx *ctx)
 }
 KEEP_PAGER(user_ta_dump_state);
 
+#ifdef CFG_TA_FTRACE_SUPPORT
+static void user_ta_dump_state_buffer(struct tee_ta_ctx *ctx, char *buf,
+				      size_t *sz)
+{
+	struct user_ta_ctx *utc = to_user_ta_ctx(ctx);
+	struct user_ta_elf *elf;
+	struct vm_region *r;
+	char flags[7] = { '\0', };
+	char desc[13];
+	size_t __maybe_unused idx = 0;
+	size_t n = 0;
+	size_t ip_size = *sz;
+	size_t write_sz = 0;
+
+	if (!buf || !ip_size)
+		return;
+
+	TAILQ_FOREACH(r, &utc->vm_info->regions, link) {
+		paddr_t pa = 0;
+
+		if (r->mobj)
+			mobj_get_pa(r->mobj, r->offset, 0, &pa);
```

Comment:
is it possible `r->mobj == 0`: and if so `pa == 0`?

Reply:
I doesn't seems to be possible scenario as I think all regions should have allocated memory objects. BTW, this function is similar to `user_ta_dump_state`, the only diff being output goes to a buffer rather than uart.